lib/src/screens/activity/widgets/activity_card.dart

```diff
@@ -30,39 +29,28 @@ class ActivityCard extends StatelessWidget {
     String subtitleTranslationKey = '';
     String? logo;
 
-    switch (logEntry.type) {
-      case LogEntryType.disclosing:
-      case LogEntryType.signing:
-        if (logEntry.serverName != null) {
-          title = logEntry.serverName!.name.translate(lang);
-          if (logEntry.serverName!.logo != null) {
-            logo = logEntry.serverName!.logo;
-          }
-        }
-        subtitleTranslationKey =
-            logEntry.type == LogEntryType.disclosing ? 'activity.data_shared' : 'activity.message_signed';
-        break;
-      case LogEntryType.issuing:
-        final cred = Credential.fromRaw(
-          irmaConfiguration: irmaConfiguration,
-          rawCredential: logEntry.issuedCredentials.first,
-        );
-        if (irmaConfiguration.issuers[cred.info.issuer.fullId] != null) {
-          title = irmaConfiguration.issuers[cred.info.issuer.fullId]!.name.translate(lang);
+    if (logEntry.type == LogEntryType.removal) {
+      final credType = irmaConfiguration.credentialTypes[logEntry.removedCredentials.keys.first]!;
+      title = irmaConfiguration.issuers[credType.fullIssuerId]!.name.translate(lang);
+      subtitleTranslationKey = 'activity.data_deleted';
+      if (credType.logo != null) {
+        logo = credType.logo;
+      }
+    } else {
+      if (logEntry.serverName != null) {
+        title = logEntry.serverName!.name.translate(lang);
+        if (logEntry.serverName!.logo != null) {
+          logo = logEntry.serverName!.logo;
         }
+      }
+
+      if (logEntry.type == LogEntryType.issuing) {
         subtitleTranslationKey = 'activity.data_received';
-        if (cred.info.credentialType.logo != null) {
-          logo = cred.info.credentialType.logo;
-        }
-        break;
-      case LogEntryType.removal:
-        final credType = irmaConfiguration.credentialTypes[logEntry.removedCredentials.keys.first]!;
-        title = irmaConfiguration.issuers[credType.fullIssuerId]!.name.translate(lang);
-        subtitleTranslationKey = 'activity.data_deleted';
-        if (credType.logo != null) {
-          logo = credType.logo;
-        }
-        break;
+      } else if (logEntry.type == LogEntryType.disclosing) {
+        subtitleTranslationKey = 'activity.data_shared';
+      } else if (logEntry.type == LogEntryType.signing) {
+        subtitleTranslationKey = 'activity.message_signed';
+      }
```

Comment:
In this session, both data is shared and data is received. We now choose to only say `data_received` on the activity card. This is not an issue being introduced in this PR, so it's fine by me if you ignore it. It might be something to discuss in the future.

![Simulator Screen Shot - iPhone 14 - 2023-01-31 at 10 48 25](https://user-images.githubusercontent.com/17035191/215726104-991a046a-7a72-4658-8fcf-1eebfaa05f52.png)

Reply:
I added both design questions to IRMA-2937